pkg/cmd/importcmd/add_collabotor.go

```diff
@@ -63,7 +63,22 @@ func (o *ImportOptions) AddAndAcceptCollaborator(newRepository bool) error {
 	// If the user creating the repo is not the pipeline user, add the pipeline user as a contributor to the repo
 	if pipelineUserName != "" && pipelineUserName != userName { // TODO: not sure why:  && o.ScmFactory.GitServerURL == o.PipelineServer {
 		// Make the invitation
-		_, _, _, err = scmClient.Repositories.AddCollaborator(ctx, fullRepoName, pipelineUserName, permission)
+		switch o.ScmFactory.GitKind {
```

Comment:
may be better to implement this in the jx-scm repo instead?

Reply:
Agree this should probably be split:
1) I found out through git blame on go-scm that the second boolean return value indicates that a user already is a collaborator - despite the semantics, I think the gitlab driver should return true if the request to add the member succeeded to indicate no further action is needed - https://github.com/jenkins-x/go-scm/blob/ca0f320a99efb8436f8ba445c9558d32df71d1e8/scm/driver/gitlab/repo.go#L259 @abayer wdyt? Still, this would require checking that return value here
2) Gitlab API requires a plain repo ID ([see doc](https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project)) while GitHub uses a reponame ([see doc](https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator)), so I'm not sure how to resolve that in go-scm when the owner is missing from the `fullRepoName`, which I understand is the case if `owner == userName` - or am I getting it wrong and `fullRepoName` always ends up having the `{owner}/{reponame}` format?

Reply:
I revisited the problem and the only thing missing here is the `owner` as part of the `fullRepoName` and to make use of the second boolean return value that indicates if a member already exists.

Proposed change in go-scm here: https://github.com/jenkins-x/go-scm/pull/299